rfcs/01924_tls_attestation_ca.md

```diff
@@ -0,0 +1,169 @@
+TLS certificate** for it, which a TEE application can then use to create TLS
+channels with clients. Such certificates are signed by the CA’s root
+certificate.
+
+While generating a TLS certificate, the CA embeds the **TEE report** (which is a
+data structure signed by the TEE platform that contains information identifying
+the code that is running inside the TEE) into it, thus binding the TLS
+certificate to the TEE firmware. This way clients can check what payload is
+being run inside the TEE they are connecting to.
+
+The TEE report also contains the hash of the **TLS public key** used by the TEE
+application. This cryptographically binds TLS credentials with the TEE report
+and prevents an attacker from replay attacks, i.e. using the same certificate
+for another TEE.
+
+As an _optional_ feature, clients can also **remotely attest** the CA itself
+before requesting its root certificate. It will create an implicit trust in the
+TEE application's code, since the CA only signs certificates for TEE
+applications that it attested itself.
+
+One of the main advantages of this remote attestation approach is that clients
+are able to use plain TLS for creating secure channels with TEE applications,
+and at the same time trust in this application without directly attesting it.
+
+There are several benefits from implementing _Remote Attestation via TLS_:
+
+- **The ability to run TLS applications as is**
+  - The proxy will allow clients to use TLS certificates to connect to TEE
+    applications
+  - Since attestation will be handled by the proxy - TEE applications could use
+    legacy TLS code without the need to support remote attestation logic
+- **Support multiple TEEs**
+  - Proxy could be dedicated to a specific TEE instance such as Intel SGX ECDSA,
+    AMD SEV-SNP, ARM Trustzone etc.
+    - It means that the client code wouldn’t need to support multiple
+      attestation protocols
+  - TEE applications will still be able to run on various TEE machines, since
+    the attestation logic is implemented by the proxy
+
+## Attestation
+
+This section presents a high-level attestation workflow overview. The workflow
+of the remote attestation involves 4 interacting entities:
+
+- **Certificate Authority (CA)**
+  - Remotely attests TEE applications
+  - Can be remotely attested by clients
+  - Creates signed certificates for TEE applications
+  - Distributes its root certificate among the clients
+  - [_Optional_] Can run inside the TEE environment
+- **Server**
+  - TEE application that processes requests from TLS clients
+  - Runs inside the TEE environment
+- **Client**
+  - Client application that connects to applications using TLS
+  - Runs on a user device or inside of a browser
+- **Provider**
+  - TEE platform provider that can confirm authenticity of TEE platform keys
+  - It’s important to note that the Provider is an external server (i.e. belongs
+    to AMD)
```

Comment:
Depending on which platforms are supported, it may not be a server. For AWS Nitro Enclaves, there is no attestation server. Instead, they have long-term keys that are used. This is, of course, a very different trust model from other platforms, and may not apply here, but it's worth considering if that model can be supported in the descriptions.